File: 2015/Day 01 Not quite LISP/go/solution.go
package main

import (
	"fmt"
	"io/ioutil"
)

func main() {
	input, _ := ioutil.ReadFile("../input.txt")
	result := processPart1(string(input))
	result2 := processPart2(string(input))
	fmt.Println("Santa takes", result, "floor")
	fmt.Println("Santa enters first the basement at", result2, "position")
}

func processPart1(input string) rune {
	var floor rune = 0

	for _, c := range input {
		if c == '(' {
			floor++
		} else {
			floor--
		}

	}
	return floor
}

func processPart2(input string) int {
	floor, position := 0, 0

	for _, c := range input {
		position++
		if c == '(' {
			floor++
		} else {
			floor--
		}
		if floor == -1 {
			break
		}
	}
	return position
}

File: 2015/Day 01 Not quite LISP/py/solution.py
import os
from colorama import Fore, Style


def processPart1(input):
    return input.count('(') - input.count(')')


def processPart2(input):
    floor = 0
    position = 0
    for i in input:
        position += 1
        floor = floor + (1 if i == '(' else -1)
        if floor == -1:
            return position


if __name__ == '__main__':
    input_path: str = os.path.abspath(
        os.path.dirname(__file__)) + '/../input.txt'
    with open(input_path) as input_file:
        data = input_file.readlines()[0]
        print('Santa takes', Fore.YELLOW +
              str(processPart1(data)) + Style.RESET_ALL, 'floor')
        print("Santa enters first the basement at", Fore.YELLOW +
              str(processPart2(data)) + Style.RESET_ALL, "position")
